src/sidebar/services/test/api-test.js

```diff
@@ -17,395 +16,283 @@ const util = require('../../../shared/test/util');
+    [
+      {
+        // Network error
+        status: -1,
+        body: 'Service unreachable.',
+        expectedMessage: 'Service unreachable.',
       },
-      [
-        {
-          // Network error
-          status: -1,
-          body: null,
-          expectedMessage: 'Service unreachable.',
-        },
-        {
-          // Request failed with an error given in the JSON body
-          status: 404,
-          statusText: 'Not found',
-          body: {
-            reason: 'Thing not found',
-          },
-          expectedMessage: '404 Not found: Thing not found',
+      {
+        // Request failed with an error given in the JSON body
+        status: 404,
+        statusText: 'Not found',
+        body: {
+          reason: 'Thing not found',
         },
-        {
-          // Request failed with a non-JSON response
-          status: 500,
-          statusText: 'Server Error',
-          body: 'Internal Server Error',
-          expectedMessage: '500 Server Error',
-        },
-      ]
-    );
-
-    it("exposes details in the Error's `response` property", function(done) {
-      api.profile.update({}, { preferences: {} }).catch(function(err) {
-        assert.match(
-          err.response,
-          sinon.match({
-            status: 404,
-            statusText: 'Not found',
-            data: {
-              reason: 'User not found',
-            },
-          })
-        );
-        done();
-      });
-      $httpBackend
-        .expectPATCH('https://example.com/api/profile')
-        .respond(function() {
-          return [404, { reason: 'User not found' }, {}, 'Not found'];
+        expectedMessage: '404 Not Found: Thing not found',
+      },
+      {
+        // Request failed with a non-JSON response
+        status: 500,
+        statusText: 'Server Error',
+        body: 'Internal Server Error',
+        expectedMessage: '500 Internal Server Error',
+      },
+    ].forEach(({ status, body, expectedMessage }) => {
+      it('rejects the call with an error', () => {
+        expectCall('patch', 'profile', status, body);
+        return api.profile.update({}, { preferences: {} }).catch(err => {
+          assert(err instanceof Error);
+          assert.equal(err.message, expectedMessage);
         });
-      $httpBackend.flush();
+      });
     });
   });
 
   it('API calls return just the JSON response if `includeMetadata` is false', () => {
-    api.profile.read({}).then(response => {
+    expectCall('get', 'profile', 200, { userid: 'acct:user@example.com' });
+    return api.profile.read({}).then(response => {
       assert.match(
         response,
         sinon.match({
           userid: 'acct:user@example.com',
         })
       );
     });
-
-    $httpBackend
-      .expectGET('https://example.com/api/profile')
-      .respond(() => [200, { userid: 'acct:user@example.com' }]);
-    $httpBackend.flush();
   });
 
   it('API calls return an `APIResponse` if `includeMetadata` is true', () => {
```

Comment:
Reading this test description makes me wonder if the `includeMetadata` confusion could be eased by renaming the property. I dunno. I keep getting caught up on it for some reason!